methods/phe-cv19-site
# Feature/create cron job to run manage tasks cv19 32

### What

Added CMS.settings.management_cron_jobs.py, which schedules management tasks (e.g. manage.py publish_scheduled_pages) using cron.
It does:
- reads cron jobs defined in management_cron_jobs.yml
- for each job, writes an executable job runner script to /code/cron_jobs which contains the complete environment for the wagtail server, as well as the command to run the management task
- adds each job to the crontab for the current user
- logs (unless told not to) the stdout and stderr for each job to /var/log/cron.log

management_cron_jobs.yml has one predefined job, which is to run publish_scheduled_pages every 30 minutes.s

### How to review

Read the code!

To test (no unit tests):
- Set the 'period' for the Publish Scheduled Pages job in management_cron_jobs.yml to something smaller (every 1 minute seems better)
- set the env up to deploy the site to an S3 bucket you can view
- schedule a page to be published sometime soon
- when the publication time is hit, you should see the html pages appear in your s3 bucket
- you should also see a log entry in /var/log/cron.log with the output from the management task
